plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/UpgradeRuntimeQuickFix.java

```diff
@@ -19,7 +19,7 @@
 /**
  * Changes runtime element to java8.
  */
-class UpgradeRuntimeQuickFix extends XsltSourceQuickFix {
+class UpgradeRuntimeQuickFix extends XsltQuickFix {
```

Comment:
This class is almost same as `UpgradeRuntimeSourceQuickFix`, except that this extends `XsltQuickFix` while the other `XsltSourceQuickFix`. Do we need both of them? What's the difference between a plain quick fix and the source quick fix?

Reply:
Yes, we do need both. They implement different interfaces and are used in different places. We have similar dualism in all the quickfixes.

Reply:
yes, you need both for different interfaces. All the quick fixes have two.